spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/rsocket/RSocketMessagingAutoConfiguration.java

```diff
@@ -36,17 +40,19 @@
  * @author Brian Clozel
  * @since 2.2.0
  */
-@Configuration(proxyBeanMethods = false)
-@ConditionalOnClass({ RSocketRequester.class, RSocketFactory.class, TcpServerTransport.class })
+@Configuration(proxyBeanMethods = true)
+@ConditionalOnClass({RSocketRequester.class, RSocketFactory.class, TcpServerTransport.class})
 @AutoConfigureAfter(RSocketStrategiesAutoConfiguration.class)
 public class RSocketMessagingAutoConfiguration {
 
+
 	@Bean
-	@ConditionalOnMissingBean
-	public RSocketMessageHandler messageHandler(RSocketStrategies rSocketStrategies) {
-		RSocketMessageHandler messageHandler = new RSocketMessageHandler();
-		messageHandler.setRSocketStrategies(rSocketStrategies);
-		return messageHandler;
+	@ConditionalOnMissingBean(RSocketMessageHandler.class)
+	public RSocketMessageHandler messageHandler(RSocketStrategies rSocketStrategies, ObjectProvider<RSocketMessageHandlerCustomizer> customizers) {
+		RSocketMessageHandlerCustomizer rSocketMessageHandlerCustomizer = customizers.getIfAvailable();
+		return rSocketMessageHandlerCustomizer.setRouteMatcher(rSocketStrategies.routeMatcher());
```

Comment:
It looks like `messageHandler.setRSocketStrategies(rSocketStrategies)` got removed as part of this change. We still want to do that and apply the customizers after that.